x-pack/plugin/stack/src/main/java/org/elasticsearch/xpack/stack/StackTemplateRegistry.java

```diff
@@ -129,7 +142,27 @@ public StackTemplateRegistry(
         NamedXContentRegistry xContentRegistry
     ) {
         super(nodeSettings, clusterService, threadPool, client, xContentRegistry);
-        this.stackTemplateEnabled = StackPlugin.STACK_TEMPLATES_ENABLED.get(nodeSettings);
+        this.clusterService = clusterService;
+        this.stackTemplateEnabled = STACK_TEMPLATES_ENABLED.get(nodeSettings);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        clusterService.getClusterSettings().addSettingsUpdateConsumer(STACK_TEMPLATES_ENABLED, this::updateEnabledSetting);
+    }
+
+    private void updateEnabledSetting(boolean newValue) {
+        if (newValue) {
+            this.stackTemplateEnabled = true;
+        } else {
+            logger.info(
+                "stack composable templates [{}] and component templates [{}] will not be installed or reinstalled",
```

Comment:
I think this is a bit confusing as to why this happened. Should we also mention that the setting was updated and as a result, the templates will not be installed/reinstalled anymore?

Reply:
the settings infrastructure already logs a message to that effect, so this looks like:

```
[2020-10-16T09:16:27,285][INFO ][o.e.c.s.ClusterSettings  ] [runTask-0] updating [stack.templates.enabled] from [true] to [false]
[2020-10-16T09:16:27,286][INFO ][o.e.x.s.StackTemplateRegistry] [runTask-0] stack composable templates [logs,metrics,synthetics] and component templates [logs-mappings,logs-settings,metrics-mappings,metrics-settings,synthetics-mappings,synthetics-settings] will not be installed or reinstalled
```

That's why I left out any mention that the setting was updated (it happens regardless)